orttraining/orttraining/python/training/onnxblock/checkpoint_utils.py

```diff
@@ -17,3 +20,18 @@ def save_checkpoint(parameters, path_to_checkpoint):
     trainable_params = [param.SerializeToString() for param in trainable_params]
     non_trainable_params = [param.SerializeToString() for param in non_trainable_params]
     _internal_save_checkpoint(trainable_params, non_trainable_params, path_to_checkpoint)
+
+
+def load_checkpoint_to_model(path_to_checkpoint, model):
+    """Loads the checkpoint to an onnx inference model."""
+
+    # Load the parameters from the checkpoint
+    parameters = _internal_load_checkpoint(path_to_checkpoint)
+
+    parameters_dict = {}
+    for (i, initializer) in enumerate(model.graph.initializer):
+        parameters_dict[initializer.name] = TensorProto()
+        parameters_dict[initializer.name].ParseFromString(parameters[i])
```

Comment:
You are making an assumption that order of initializers and parameters loaded from the checkpoint is same. This may not be true all the time.

Reply:
+1.